src/Chainweb/CutDB.hs

```diff
@@ -420,6 +430,16 @@ startCutDb config logfun headerStore payloadStore cutHashesStore = mask_ $ do
                             -- FIXME: this requires that the pact db is deleted because it
                             -- interfers with rewind limit. As a fix temporarily disable
                             -- rewind limit during initialization.
+    readCutForHeight h wbhdb' = do
+      unsafeMkCut v . HM.fromList <$> (flip traverse (HS.toList $ chainIdsAt v h) $ \cid -> do
+            db <- getWebBlockHeaderDb wbhdb' cid
+            let randomStrElem s = do
+                  l <- S.length_ s
+                  i <- createSystemRandom >>= asGenIO (uniformR (0,l-1))
```

Comment:
This line should be 
```
i <- createSystemRandom >>= asGenIO (uniformR (0,l-2))
```
But I know someone else is going to take this over. I'm just putting this here as an FYI.